internal/brokers/manager_test.go

```diff
@@ -0,0 +1,209 @@
+
+			// Grab the list of broker names from the manager to use as golden file.
+			var brokers []string
+			for _, broker := range got.AvailableBrokers() {
+				brokers = append(brokers, broker.Name)
+			}
+
+			want := testutils.LoadWithUpdateFromGoldenYAML(t, brokers)
+			require.Equal(t, want, brokers, "NewManager should return the expected brokers, but did not")
+		})
+	}
+}
+
+func TestBrokerForSessionID(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		sessionID string
+
+		wantBroker string
+		wantErr    bool
+	}{
+		"Successfully returns expected broker":         {sessionID: "success"},
+		"Returns local broker when sessionID is empty": {wantBroker: "local"},
+
+		"Error when there is no broker for sessionID": {sessionID: "does not exist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			b := newBrokerForTests(t)
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			if tc.wantBroker == "" {
+				tc.wantBroker = strings.ReplaceAll(t.Name(), "/", "_")
+			}
+			if tc.sessionID != "does not exist" {
+				m.SetBrokerForSessionID(tc.sessionID, &b)
+			}
+
+			got, err := m.BrokerForSessionID(tc.sessionID)
+			if tc.wantErr {
+				require.Error(t, err, "BrokerForSessionID should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "BrokerForSessionID should not return an error, but did")
+
+			require.Equal(t, tc.wantBroker, got.Name, "BrokerForSessionID should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestGetBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		brokerID string
+
+		wantBroker string
+		wantErr    bool
+	}{
+		"Successfully returns expected broker": {brokerID: "local", wantBroker: "local"},
+
+		"Error when broker does not exist": {brokerID: "does not exist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			got, err := m.GetBroker(tc.brokerID)
+			if tc.wantErr {
+				require.Error(t, err, "GetBroker should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "GetBroker should not return an error, but did")
+			require.Equal(t, tc.wantBroker, got.ID, "GetBroker should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
+	t.Parallel()
+
+	m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+	require.NoError(t, err, "Setup: could not create manager")
+
+	brokers := m.AvailableBrokers()
+	//#nosec // This just to vary the selected broker in the tests, we don't care about randomness safety.
```

Comment:
just a niptick: I used //nolint:gosec or something like this in the other PR, maybe we should align on this which I feel is more goish right now, wdyt?